### PR TITLE
use `sync.Once` to init the global tuf root

### PR DESCRIPTION
**Pull request by dekkagaijin**
Follow-up to https://github.com/sigstore/cosign/pull/1158

```release-note
NONE
```

### CODE REVIEW

File: pkg/cosign/tuf/client.go [hunk 1]
```diff
@@ -129,101 +129,105 @@ func isExpiredMetadata(metadata []byte) bool {
...
 	}
 
+	rootClient = uninitializedClient
 	return rootClient, nil
 }
 
+// Gets the global TUF client if the directory exists.
+// This will not make a remote call unless fetch is true.
+func RootClient(ctx context.Context, remote client.RemoteStore, altRoot []byte) (*client.Client, error) {
+	var err error
```
Comment (mattmoor) [inline pkg/cosign/tuf/client.go:222]: This needs to be a global, or subsequent calls will return `nil, nil`

File: pkg/cosign/tuf/client.go [hunk 2]
```diff
@@ -129,101 +129,105 @@ func isExpiredMetadata(metadata []byte) bool {
...
-						return nil, errors.Wrap(err, "setting local meta")
-					}
+				if err := local.SetMeta(mdFilename, msg); err != nil {
+					return nil, errors.Wrap(err, "setting local meta")
 				}
-				rootClient = client.NewClient(local, remote)
-				return rootClient, nil
 			}
+			rootClient = client.NewClient(local, remote)
+			return rootClient, nil
```
Comment (mattmoor) [inline pkg/cosign/tuf/client.go:170]: This shouldn't touch `rootClient`

File: pkg/cosign/tuf/client.go [hunk 3]
```diff
@@ -129,101 +129,105 @@ func isExpiredMetadata(metadata []byte) bool {
...
+		if err := uninitializedClient.Init(rootKeys, rootThreshold); err != nil {
+			return nil, errors.Wrap(err, "initializing root client")
+		}
+		if err := updateMetadataAndDownloadTargets(uninitializedClient); err != nil {
+			return nil, errors.Wrap(err, "updating from remote TUF repository")
+		}
 	}
 
+	rootClient = uninitializedClient
 	return rootClient, nil
```
Comment (mattmoor) [inline pkg/cosign/tuf/client.go:216]: same, no touchy rootClient

File: pkg/cosign/tuf/client.go (hunk 1 above)
Comment (dekkagaijin) [reply pkg/cosign/tuf/client.go]: fixed

File: pkg/cosign/tuf/client.go (hunk 3 above)
Comment (dekkagaijin) [reply pkg/cosign/tuf/client.go]: fixed

**Review by mattmoor — APPROVED** [review]

**Pull request merged**